packages/deployments/contracts/contracts/core/connext/facets/SwapAdminFacet.sol

```diff
@@ -176,6 +185,34 @@ contract SwapAdminFacet is BaseConnextFacet {
     emit SwapInitialized(_key, entry, msg.sender);
   }
 
+  /**
+   * @notice remove Swap Struct for key
+   *
+   * @param _key the hash of the canonical id and domain for token
+   */
+  function removeSwap(bytes32 _key) external onlyOwnerOrAdmin {
+    uint8 numPooledTokens = uint8(s.swapStorages[_key].pooledTokens.length);
+
+    if (numPooledTokens == 0) revert SwapAdminFacet__removeSwap_notInitialized();
+
+    for (uint8 i; i < numPooledTokens; ) {
+      if (s.swapStorages[_key].balances[i] > 0) {
+        revert SwapAdminFacet__removeSwap_NonZeroBalance();
+      }
+
+      if (s.swapStorages[_key].adminFees[i] > 0) {
+        s.swapStorages[_key].withdrawAdminFees(msg.sender);
+      }
+
+      unchecked {
+        ++i;
+      }
+    }
+
+    delete s.swapStorages[_key];
```

Comment:
I think this `delete` operation won't clear the nested array properties in `Swap`, so removing a swap and recreating one with the same key could result in reusing old references (such as for `IERC20[] pooledTokens`) unless they are explicitly overwritten when we do the recreation.

In the loop above (`for (uint8 i; i < numPooledTokens; ) {`) we should probs just delete entries in all of the nested arrays (pooledTokens, tokenPrecisionMultipliers, balances, and adminFees) to make sure everything is cleaned out.